pkg/cloud/services/ssm/secret.go

```diff
@@ -127,3 +150,17 @@ func (s *Service) Delete(m *scope.MachineScope) error {
 
 	return kerrors.NewAggregate(errs)
 }
+
+// MapToSSMTags converts infrav1.Tags (a map of string key-value pairs) to a slice of SSM Tag objects.
+func MapToSSMTags(tags infrav1.Tags) []types.Tag {
```

Comment:
Does this need to be a public function?

Reply:
Also it would be nice to add a unit test for this function in the _test.go file.

Reply:
Thanks @damdo , I think it is safer and ok to not have it as public func in my understanding, will change that , will add a unit test as well.

Reply:
Upon referring other PRs example- #5498 I got to find the converters package most suitable and added unit test as well .